homeassistant/components/twinkly/light.py

```diff
@@ -224,25 +211,24 @@ async def async_update(self) -> None:
                 DEV_NAME in device_info
                 and DEV_MODEL in device_info
                 and (
-                    device_info[DEV_NAME] != self.__name
-                    or device_info[DEV_MODEL] != self.__model
+                    device_info[DEV_NAME] != self._name
+                    or device_info[DEV_MODEL] != self._model
                 )
             ):
-                self.__name = device_info[DEV_NAME]
-                self.__model = device_info[DEV_MODEL]
-
-                if self._conf is not None:
-                    # If the name has changed, persist it in conf entry,
-                    # so we will be able to restore this new name if hass is started while the LED string is offline.
-                    self.hass.config_entries.async_update_entry(
-                        self._conf,
-                        data={
-                            CONF_ENTRY_HOST: self._client.host,  # this cannot change
-                            CONF_ENTRY_ID: self._id,  # this cannot change
-                            CONF_ENTRY_NAME: self.__name,
-                            CONF_ENTRY_MODEL: self.__model,
-                        },
-                    )
+                self._name = device_info[DEV_NAME]
+                self._model = device_info[DEV_MODEL]
+
+                # If the name has changed, persist it in conf entry,
+                # so we will be able to restore this new name if hass is started while the LED string is offline.
+                self.hass.config_entries.async_update_entry(
+                    self._conf,
+                    data={
+                        ENTRY_DATA_HOST: self._client.host,  # this cannot change
+                        ENTRY_DATA_ID: self._id,  # this cannot change
+                        ENTRY_DATA_NAME: self._name,
+                        ENTRY_DATA_MODEL: self._model,
+                    },
+                )
 
             for key, value in device_info.items():
                 if key not in HIDDEN_DEV_VALUES:
```

Comment:
Do we know all the keys that may exist in device_info? We want to see what they are and their formatting. Can we invert this check and list the allowed keys explicitly?

Reply:
I saw your comment about that as well in #42103. At the moment I have no clue what's in there. It needs some more digging. I would propose to address that separatly.

Reply:
Ok, sounds good. 👍